### PR TITLE
tests/crm: Marvell specific changes

### PR DESCRIPTION
**Pull request by sureshsekar28**
As per Marvell design nexthop increments only after adding route
Signed-off-by: sureshsekar28 <sureshsekar@marvell.com>

### Description of PR
As per Marvell design nexthop increments only after adding route

Summary:
Fixes # As per Marvell design nexthop increments only after adding route

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ x] Test case(new/improvement)

### Back port request
- [x ] 201911
- [ x] 202012
- [x ] 202205

#### What is the motivation for this PR?
As per Marvell design nexthop increments only after adding route
#### How did you do it?
script enhancement
#### How did you verify/test it?
script passing with the change
#### Any platform specific information?
Marvell specific changes
### Documentation
NA

### CODE REVIEW

**Comment by Blueve** [comment]
Do we need this fix to 201911 and 202012?

**Comment by sureshsekar28** [comment]
Yes

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Jing Kan ***@***.***>
Sent: Thursday, December 8, 2022 7:21:28 PM
To: sonic-net/sonic-mgmt ***@***.***>
Cc: Suresh Sekar ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)

External Email
________________________________

Do we need this fix to 201911 and 202012?

File: tests/crm/test_crm.py [hunk 1]
```diff
@@ -552,29 +552,49 @@ def test_crm_route(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_fro
...
+def test_crm_nexthop(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index, crm_interface, ip_ver, nexthop, ptfhost):
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
     asichost = duthost.asic_instance(enum_frontend_asic_index)
     RESTORE_CMDS["crm_threshold_name"] = "ipv{ip_ver}_nexthop".format(ip_ver=ip_ver)
-    nexthop_add_cmd = "{ip_cmd} neigh replace {nexthop} \
+    if duthost.facts["asic_type"] == "marvell":
+        vlan_mem_del_cmd = "config vlan member del 1000 Ethernet1"
+        vlan_mem_add_cmd =  "config vlan member add 1000 Ethernet1 -u"
+        if ip_ver == "4":
+           ptf_ip_add_cmd = "ifconfig eth1 {}/24 up".format(nexthop)
```
Comment (Blueve) [inline tests/crm/test_crm.py:563]: How about use existing ptf method:
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/ptf.py#L51
and 
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/ptf.py#L90

File: tests/crm/test_crm.py [hunk 2]
```diff
@@ -552,29 +552,49 @@ def test_crm_route(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_fro
...
+           nexthop_del_cmd = "config route del prefix 99.99.99.0/24 nexthop {}".format(nexthop)
+        else:
+           ptf_ip_add_cmd = "ifconfig eth1 inet6 add {}/96".format(nexthop)
+           ptf_ip_rm_cmd = "ifconfig eth1 inet6 del {}/96".format(nexthop)
+           ptfhost.shell(ptf_ip_add_cmd)
+           ip_add_cmd = "config interface ip add Ethernet1 2001::2/64"
+           ip_remove_cmd = "config interface ip remove Ethernet1 2001::2/64"
+           nexthop_add_cmd = "config route add prefix 3001::0/64 nexthop {}".format(nexthop)
+           nexthop_del_cmd = "config route del prefix 3001::0/64 nexthop {}".format(nexthop)
+        asichost.shell(vlan_mem_del_cmd)
```
Comment (Blueve) [inline tests/crm/test_crm.py:578]: How about use sonic host method?
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/sonic.py#L2078
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/sonic.py#L2068

File: tests/crm/test_crm.py (hunk 1 above)
Comment (sureshsekar28) [reply tests/crm/test_crm.py:563]: i dont find add_ip_to_dev method,
image

File: tests/crm/test_crm.py (hunk 2 above)
Comment (sureshsekar28) [reply tests/crm/test_crm.py:578]: I dont find del_member_from_vlan method,
(Pdb) asichost.del_member_from_vlan(1000,'Ethernet1')
*** AttributeError: 'SonicAsic' object has no attribute 'del_member_from_vlan'
(Pdb) dir(asichost)
['_MULTI_ASIC_DOCKER_NAME', '_MULTI_ASIC_SERVICE_NAME', '__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'asic_index', 'bgp_drop_rule', 'bgp_facts', 'check_bgp_statistic', 'cli_ns_option', 'command', 'config_facts', 'config_ip_intf', 'config_portchannel', 'config_portchannel_member', 'create_ssh_tunnel_sai_rpc', 'delete_container', 'get_active_ip_interfaces', 'get_asic_namespace', 'get_bgp_statistic', 'get_critical_services', 'get_default_route_from_app_db', 'get_docker_cmd', 'get_docker_name', 'get_extended_minigraph_facts', 'get_ip_route_info', 'get_portchannel_and_members_in_ns', 'get_queue_oid', 'get_router_mac', 'get_service_name', 'interface_facts', 'ip_cmd', 'is_backend_portchannel', 'is_container_running', 'is_default_route_removed_from_app_db', 'is_it_backend', 'is_it_frontend', 'is_service_running', 'namespace', 'ns_arg', 'os_version', 'ping_v4', 'port_exists', 'port_on_asic', 'portchannel_on_asic', 'ports', 'queue_oid', 'remove_ssh_tunnel_sai_rpc', 'reset_service', 'restart_service', 'run_redis_cli_cmd', 'run_redis_cmd', 'run_sonic_db_cli_cmd', 'run_vtysh', 'shell', 'show_interface', 'show_ip_interface', 'shutdown_interface', 'sonic_db_cli', 'sonic_release', 'sonichost', 'start_service', 'startup_interface', 'stop_service', 'switch_arptable']

File: tests/crm/test_crm.py (hunk 1 above)
Comment (sureshsekar28) [reply tests/crm/test_crm.py]: Thanks, addressed review comments

File: tests/crm/test_crm.py (hunk 2 above)
Comment (sureshsekar28) [reply tests/crm/test_crm.py]: Thanks,addressed review comments

**Review by Blueve — APPROVED** [review]

**Review by Blueve — APPROVED** [review]

**Comment by Blueve** [comment]
Hi @sureshsekar28, there are some new styling issues introduced by your change, can you help fix them?
```
tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)
tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:561:40: E231 missing whitespace after ','
tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:562:45: E231 missing whitespace after ','
tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:40: E231 missing whitespace after ','
tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:569:45: E231 missing whitespace after ','
tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:610:51: E231 missing whitespace after ','
tests/crm/test_crm.py:610:63: E231 missing whitespace after ','
```

**Review by Blueve — CHANGES_REQUESTED** [review]
Hi @sureshsekar28, there are some new styling issues introduced by your change, can you help fix them?
```
tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)
tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:561:40: E231 missing whitespace after ','
tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:562:45: E231 missing whitespace after ','
tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:40: E231 missing whitespace after ','
tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:569:45: E231 missing whitespace after ','
tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:610:51: E231 missing whitespace after ','
tests/crm/test_crm.py:610:63: E231 missing whitespace after ','
```

**Comment by sureshsekar28** [comment]
Thanks. I have corrected styling issues. Please check now.

From: Jing Kan ***@***.***>
Date: Monday, 12 December 2022 at 2:01 PM
To: sonic-net/sonic-mgmt ***@***.***>
Cc: Suresh Sekar ***@***.***>, Mention ***@***.***>
Subject: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)
External Email
________________________________

@Blueve requested changes on this pull request.

Hi @sureshsekar28<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sureshsekar28&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=51fdelZPX1e7xIFA3Ql8AkllMJ3yuGrbS-4FnowMy94&e=>, there are some new styling issues introduced by your change, can you help fix them?

tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)

tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:561:40: E231 missing whitespace after ','

tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:562:45: E231 missing whitespace after ','

tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:40: E231 missing whitespace after ','

tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:569:45: E231 missing whitespace after ','

tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:610:51: E231 missing whitespace after ','

tests/crm/test_crm.py:610:63: E231 missing whitespace after ','

**Comment by sureshsekar28** [comment]
I have addressed all the review comments. Please review.

Thanks,
Suresh.

From: Suresh Sekar ***@***.***>
Date: Monday, 12 December 2022 at 3:07 PM
To: sonic-net/sonic-mgmt ***@***.***>, sonic-net/sonic-mgmt ***@***.***>
Cc: Mention ***@***.***>
Subject: Re: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)
Thanks. I have corrected styling issues. Please check now.

From: Jing Kan ***@***.***>
Date: Monday, 12 December 2022 at 2:01 PM
To: sonic-net/sonic-mgmt ***@***.***>
Cc: Suresh Sekar ***@***.***>, Mention ***@***.***>
Subject: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)
External Email
________________________________

@Blueve requested changes on this pull request.

Hi @sureshsekar28<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sureshsekar28&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=51fdelZPX1e7xIFA3Ql8AkllMJ3yuGrbS-4FnowMy94&e=>, there are some new styling issues introduced by your change, can you help fix them?

tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)

tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:561:40: E231 missing whitespace after ','

tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:562:45: E231 missing whitespace after ','

tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:40: E231 missing whitespace after ','

tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:569:45: E231 missing whitespace after ','

tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:610:51: E231 missing whitespace after ','

tests/crm/test_crm.py:610:63: E231 missing whitespace after ','

**Review by Blueve — APPROVED** [review]

**Pull request merged**